x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextField.java

```diff
@@ -186,6 +194,140 @@ private void validateFieldNotPresent(String field, Object fieldValue) {
         }
     }
 
+    public abstract static class IndexOptions implements ToXContentObject {
+
+        private final String type;
+
+        public IndexOptions(String type) {
+            this.type = type;
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            for (Map.Entry<String, Object> entry : asMap().entrySet()) {
+                if (entry.getValue() != null) {
+                    builder.field(entry.getKey(), entry.getValue());
+                }
+            }
+            builder.endObject();
+            return builder;
+        }
+
+        public String type() {
+            return type;
+        }
+
+        @Override
+        public String toString() {
+            return asMap().toString();
+        }
+
+        public Map<String, Object> asMap() {
+            Map<String, Object> map = new HashMap<>();
+            if (type != null) {
+                map.put(TYPE_FIELD, type);
+            }
+
+            map.putAll(
+                addParamsToMap().entrySet()
+                    .stream()
+                    .filter(entry -> entry.getValue() != null)
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))
+            );
+            return map;
+        }
+
+        public abstract Map<String, Object> addParamsToMap();
+
+        public abstract void validate(String fieldName);
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            IndexOptions that = (IndexOptions) o;
+            return Objects.equals(type, that.type);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(type);
+        }
+    }
+
+    public static class DenseVectorIndexOptions extends IndexOptions {
```

Comment:
Can we avoid duplicating this logic from `DenseVectorFieldMapper.IndexOptions`? I don't understand why we need parsing and validation logic on top of what's already exposed in the `DenseVectorFieldMapper.IndexOptions` definitions.